src/Glpi/Form/Migration/FormMigration.php

```diff
@@ -1380,29 +1380,52 @@ private function processMigrationOfVisibilityConditions(): void
                     json_decode($question->fields['extra_data'], true)
                 );
                 $condition_handlers = $question_type->getConditionHandlers($config);
-                $condition_handler   = null;
 
                 // Get the value operator before trying to find a compatible handler
                 $value_operator = $this->getValueOperatorFromLegacy($raw_condition['show_condition']);
 
-                if ($value_operator !== null) {
-                    $condition_handler = current(array_filter(
-                        $condition_handlers,
-                        function (ConditionHandlerInterface $handler) use ($value_operator) {
-                            if (!($handler instanceof ConditionHandlerDataConverterInterface)) {
-                                return false;
-                            }
-
-                            return in_array(
-                                $value_operator,
-                                $handler->getSupportedValueOperators()
-                            );
-                        }
-                    ));
+                // If the value operator is null, we skip this condition
+                if ($value_operator === null) {
+                    continue;
+                }
+
+                $condition_handler = current(array_filter(
+                    $condition_handlers,
+                    fn(ConditionHandlerInterface $handler) => in_array(
+                        $value_operator,
+                        $handler->getSupportedValueOperators()
+                    )
+                ));
+
+                // If no condition handler is found for the value operator, we skip this condition
+                if ($condition_handler === false) {
+                    /** @var Question|Comment|Section|FormDestination $item */
+                    $item = getItemForItemtype($target_item['itemtype']);
+                    if ($item->getFromDB($target_item['items_id']) === false) {
+                        throw new LogicException(
+                            sprintf(
+                                'Target item %s with ID %d not found for visibility condition.',
+                                $target_item['itemtype'],
+                                $target_item['items_id']
+                            )
+                        );
```

Comment:
This condition will be skipper anyway, I think it is preferable to not block the migration with an exception in this case. For instance, you can use fallback values:
```php
                        sprintf(
                            __('A visibility condition used in "%s" "%s" (Form "%s") with value operator "%s" is not supported by the question type. It will be ignored.'),
                            $item->getTypeName(1) ?? $target_item['itemtype'],
                            $item->getName() ?? $target_item['items_id'],
                            $item->getForm()->getName() ?? NOT_AVAILABLE,
                            $value_operator->getLabel()
                        )
```